File: C#/Src/MethodParameter.cs
using System;
using Newtonsoft.Json;

namespace iKnodeSdk
{
    /// <summary>
    /// Defines the Method Parameter.
    /// </summary>
    public class MethodParameter
    {
		/// <summary>
		/// Gets or Sets the Parameter Name.
		/// </summary>
		/// <value>Parameter Name.</value>
		public string Name
		{
			get;
			set;
		}

		/// <summary>
		/// Gets or Sets the Parameter Value.
		/// </summary>
		/// <value>Parameter Value.</value>
		public object Value
		{
			get;
			set;
		}

        /// <summary>
        /// Initializes a new instance of the <see cref="MethodParameter"/> class.
        /// </summary>
        /// <param name="name">Parameter Name.</param>
        /// <param name="value">Parameter Value.</param>
		public MethodParameter(string name, object value)
		{
			this.Name = name;
			this.Value = value;
		}

        /// <summary>
        /// Initializes a new instance of the <see cref="MethodParameter"/> class.
        /// </summary>
        /// <param name="name">Parameter Name.</param>
        /// <param name="value">Parameter Value.</param>
        /// <param name="valueType">Value Type.</param>
        public MethodParameter(string name, string value, string valueType)
		{
			this.Name = name;
			this.Value = value;

            this.Value = JsonConvert.DeserializeObject(value, Type.GetType(valueType));
		}
    }
}

File: Java/src/src/iKnodeSdk/MethodParameter.java
package iKnodeSdk;

/**
 * Defines the MethodParameter.
 * 
 * @author jgemedina
 * @version 0.1
 */
public final class MethodParameter {
	/**
	 * Parameter name.
	 * 
	 * @since 0.1
	 */
	private String _name;
	
	/**
	 * Parameter value.
	 * 
	 * @since 0.1
	 */
	private Object _value;

	/**
	 * Gets the Parameter name.
	 * 
	 * @return Parameter name.
	 * @since 0.1
	 */
	public String getName() {
		return _name;
	}

	/**
	 * Sets the Parameter name.
	 * 
	 * @param _name Parameter name.
	 * @since 0.1
	 */
	public void setName(String _name) {
		this._name = _name;
	}

	/**
	 * Gets the Parameter value.
	 * 
	 * @return Parameter value.
	 * @since 0.1
	 */
	public Object getValue() {
		return _value;
	}

	/**
	 * Sets the Parameter value.
	 * 
	 * @param value Parameter value.
	 * @since 0.1
	 */
	public void setValue(Object value) {
		this._value = value;
	}
	
	/**
	 * Initializes a new instance of {@link MethodParameter} class.
	 * 
	 * @param name Parameter name.
	 * @param value Parameter value.
	 * @since 0.1
	 */
	public MethodParameter(final String name, final Object value)
	{
		this._name = name;
		this._value = value;
	}
}